Sources/AppcuesKit/Data/Models/Experience.swift

```diff
@@ -115,6 +153,18 @@ extension Experience: Decodable {
         case redirectURL = "redirectUrl"
         case nextContentID = "nextContentId"
     }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        id = try container.decode(UUID.self, forKey: .id)
+        name = try container.decode(String.self, forKey: .name)
+        type = try container.decode(String.self, forKey: .type)
+        publishedAt = try? container.decode(Int.self, forKey: .publishedAt)
```

Comment:
Interesting thing I discovered recently: In Xcode 14, you can cmd+click on a Codable type and choose `Add Explicit Codable Implementation`. For something like `let optionalValue: String?`, it generates `try container.decodeIfPresent` rather than `try? container.decode`. I wasn't familiar with `decodeIfPresent`, but the [docs](https://developer.apple.com/documentation/swift/keyeddecodingcontainer/decodeifpresent(_:forkey:)-2hn6i) are insightful:

> A decoded value of the requested type, or nil if the Decoder does not have an entry associated with the given key, or if the value is a null value.

So the nuance being that JSON where the key exists with the wrong value type (eg `"optionalValue": true`) would fail the decoding with an error for `try container.decodeIfPresent` but just set a nil value with `try? container.decode` because `try?` swallows the error. Essentially, `decodeIfPresent` is a stronger check than `try? container.decode`.

The default `Decodable` conformance before this change uses `decodeIfPresent`, so technically this is weakening the model decoding validation. I wouldn't say that's a problem (or that this needs to be changed), but it is something to be aware of with all the usages of `try?` in this PR.

As another example I just looked at the places where we currently use `try?` in decoders, and there's `qualificationReason = try? container.decode(QualificationReason.self, forKey: .qualificationReason)` where `try?` would be the right usage because don't want an error for unknown qualificationReason values.

Reply:
That is good to understand the difference here in the options - thanks for the info! I scanned through `try?` usages here again and didn't really see any I'd want stricter behavior on either. But let me know if anything else jumps out that seems worth of improving along those lines.

Reply:
Agreed that `try?` is appropriate for the uses in this PR 👍